src/ANSITerminal_unix.ml

```diff
@@ -175,26 +175,28 @@ let style_to_string = function
   | Background White -> "47"
   | Background Default -> "49"
 
-
-let print_with pr ~tty style txt =
+let print_with style txt oc =
```

Comment:
I do not mind you modify earlier code as part of your PR but could you please split it in two commits: one for existing code modifications — including moving printf functions to their own sub-module — and one adding Format support.